include/swift/AST/Pattern.h

```diff
@@ -191,6 +191,9 @@ class alignas(8) Pattern {
   /// Does this binding declare something that requires storage?
   bool hasStorage() const;
 
+  /// Can this binding be initialized with a default value?
+  bool isDefaultInitializable() const;
```

Comment:
This is the negation of `isNeverDefaultInitializable` which is used across multiple files and had to be extracted. `isNeverDefaultInitializable` seemed strange to expose, though; if there's a better name for this, I think we should use it. I don't think that negation is strong enough to promise default initializability...